packages/echo/src/client.ts

```diff
@@ -496,6 +497,13 @@ export class Echo {
 
   private executeStepFactory<T, U>(event: IEvent, setResult: (result: any) => void): ActionStep<T, U> {
     return async (stepId, stepResolve, options) => {
+      if (await this.shouldSkip(options?.skip, event.data)) {
```

Comment:
i don't see a reason to execute the step is we will skip the step.

Reply:
Just to confirm, if a step is marked as skipped, should we also skip its execution?

Reply:
Yes, sorry if my comment caused confusion - we're aligned that we should skip execution `skip` function returns `true`.